qutebrowser/components/misccommands.py

```diff
@@ -339,16 +339,23 @@ def debug_webaction(tab: apitypes.Tab, action: str, count: int = 1) -> None:
 
 @cmdutils.register()
 @cmdutils.argument('tab', value=cmdutils.Value.count_tab)
-def tab_mute(tab: Optional[apitypes.Tab]) -> None:
+@cmdutils.argument('to_set', choices=['true', 'false'])
+def tab_mute(tab: Optional[apitypes.Tab], to_set: str = None) -> None:
```

Comment:
I'm pretty sure you can have `to_set: Optional[bool] = None` to make the cmdutils machinery take care of validating and parsing the argument for you. Then you should be able to get rid of the `choices` kwarg above (in fact I think you might be able to get rid of the whole cmdutils.argument() decorator) and treat it as a bool within the method, and have less conditionals hopefully.

Reply:
That doesn't seem to work, unfortunately. I don't think we have any other place where we have a bool as a positional argument, bools are usually converted to flags otherwise.

Ideally this would probably be fixed in the cmdutils code, but if @Dleavenworth would rather not look at that (it might be a bit more complex to get into), I'm fine with doing that as a follow-up issue.

Reply:
I could try looking into it. When I was initially implementing this, I tried a boolean, but ran into Python's bool() type conversion function having some rather interesting behavior. Any suggestions on how to tackle this?

Reply:
So, Python's `bool` does not interpret any string content - all it does is basically:

```python
if s:
    return True
else:
    return False
```

Which means that the same thing happens as with any other container (e.g. a list): An empty string is converted to `False`, but any other string is converted to `True`.

However, for an user-facing value, you'd typically want things like `yes`, `true`, `on` to count as `True`, and `no`, `false`, `off` to count as `False`. One way this can be done is via a dict, and we do in fact already have something like this for config values: 

https://github.com/qutebrowser/qutebrowser/blob/951a14a66bb39dee089887642ac4c2b6dcb3b612/qutebrowser/config/configtypes.py#L78-L80

https://github.com/qutebrowser/qutebrowser/blob/951a14a66bb39dee089887642ac4c2b6dcb3b612/qutebrowser/config/configtypes.py#L753-L756

So I'd approach this by:

- Moving this code into a (tiny) function in `utils.py`, and perhaps adding a test or two for it too
- Calling that from `configtypes.py` instead
- Find out what needs to change in `qutebrowser/commands/argparser.py` and `qutebrowser/commands/command.py` to call this function to convert boolean positional arguments.
  - `Command._get_param_value` will probably need to change - right now it has a `elif typ is bool:  # no type conversion for flags`, that would probably need to change to check whether it's a positional parameter, or a keyword-only one (keyword-only would still be a flag)
  - There is also some `is_bool` stuff going on there, which might need some changes
  - Perhaps some command handlers currently use positional arguments, but actually expect having a flag? Those might need to be changed to keyword-only.
  - Finally, those changes should have some tests added to `tests/unit/commands/`

Reply:
Thanks for the suggestions! I'll see what I can do